packages/wrangler/src/pages/functions/buildWorker.ts

```diff
@@ -164,3 +148,135 @@ export function buildWorker({
 		}
 	);
 }
+
+export type RawOptions = {
+	workerScriptPath: string;
+	outfile: string;
+	directory: string;
+	minify?: boolean;
+	sourcemap?: boolean;
+	watch?: boolean;
+	plugins?: Plugin[];
+	onEnd?: () => void;
+	buildOutputDirectory?: string;
+	nodeCompat?: boolean;
+	local: boolean;
+	betaD1Shims?: string[];
+};
+
+/**
+ * This function bundles a raw `_worker.js` Pages file
+ * before it gets deployed.
+ *
+ * This allows Wrangler to add shims and other wrappers
+ * around the handlers, which is useful to support beta features.
+ */
+export function buildRawWorker({
```

Comment:
IMHO, unless we plan to use this fn for other scripts than just `_worker.js`, we should call this `buildWorkerJS`. Most of our codebase refers to Advanced Mode's `_worker.js` as `_workerJS`, to the point that, I'd argue, it has become our naming convention for this file. I'd +1 to staying consistent.

Furthermore, I'd argue that with bundling _worker.js becoming a thing, we might want to rename the existing `buildWorker` fn to smth like `buildFunctionsWorker`, and put a comment on it just like you did with `buildRawWorker`, to make the code more readable. Outside the scope of this PR tho, will add a note for myself to address this in a separate PR

Reply:
since this needs to fit a more generic use case than just `_worker.js` (`.ts`/`.jsx`, etc), let's leave it as is for now